lib/hooks/userspace/hook-grpc.js

```diff
@@ -19,14 +19,71 @@
 var cls = require('../../cls.js');
 var shimmer = require('shimmer');
 var semver = require('semver');
+var SpanData = require('../../span-data.js');
+
 var agent;
 
 var SUPPORTED_VERSIONS = '0.13 - 0.15';
 
-function startBatchWrap(startBatch) {
-  return function startBatchTrace(thing, callback) {
-    // TODO: maybe we only want to do this if a root context exists.
-    return startBatch.call(this, thing, cls.getNamespace().bind(callback));
+function wrapClientMethod(name, method, class_options) {
+  return function() {
```

Comment:
I think we will want to have this function named so the stack traces are meaningful.